solver/app/v2/tokens.go

```diff
@@ -0,0 +1,110 @@
+package appv2
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/evmchain"
+	"github.com/omni-network/omni/lib/netconf"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// TODO: consider moving this to lib/tokens
+
+type Token struct {
+	symbol      string
+	name        string
+	chainID     uint64
+	address     common.Address // empty if native
+	coingeckoID string
+}
+
+type Expense struct {
+	token  Token
+	amount *big.Int
+}
+
+type Tokens []Token
+
+func (t Token) isNative() bool {
```

Comment:
same here, private functions should not be used by other types (even in same package).